qutebrowser/utils/utils.py

```diff
@@ -864,6 +864,63 @@ def unused(_arg):
     pass
 
 
+def parse_numsettxt_into_list(txtset, nummax):
```

Comment:
Yeah, I was pretty much thinking of that when I wrote `is_numbers_in_numsettxt`, with the only exception being ordering the ranges, so that (after the ordering) I can apply a small trick so that the worst run time complexity becomes `O(max(d,r))` instead of `O(d*r)`.

Does Python 3 use Timsort? If so, then probably that's the bottleneck now in terms of asymptotic worst time complexity analysis. I think, after considering Timsort's complexity, the overall complexity is `O(max(d log d, r log r))`, which is practically much better than `O(len(possibly-crazy user-defined index set))`.

Now that the problem is Timsort, I can now rest the time complexity case :-p.

I'll shortly commit a version with these changes. I did not use `range(...)` because I wanted to sort the intervals to avoid `O(d*r)` and achieve `O(max(d log d, r log r))` instead. As far as I tried, it seems to me that lists of ranges cannot be sorted :man_shrugging:.